pkg/inst-api/instrumenter/span_supressor.go

```diff
@@ -65,4 +66,30 @@ func (s *SpanKeySuppressor) ShouldSuppress(parentContext context.Context, spanKi
 	return true
 }
 
-// TODO: semconv span suppressor
+func NewSpanKindSuppressor() *SpanKindSuppressor {
+	var m = make(map[trace.SpanKind]SpanSuppressor)
+	m[trace.SpanKindServer] = NewSpanKeySuppressor([]attribute.Key{utils.KIND_SERVER})
+	m[trace.SpanKindClient] = NewSpanKeySuppressor([]attribute.Key{utils.KIND_CLIENT})
+	m[trace.SpanKindProducer] = NewSpanKeySuppressor([]attribute.Key{utils.KIND_PRODUCER})
+	m[trace.SpanKindConsumer] = NewSpanKeySuppressor([]attribute.Key{utils.KIND_CONSUMER})
+
+	return &SpanKindSuppressor{
+		delegates: m,
+	}
+}
+
+func (s *SpanKindSuppressor) StoreInContext(context context.Context, spanKind trace.SpanKind, span trace.Span) context.Context {
+	spanSuppressor, exists := s.delegates[spanKind]
+	if !exists {
```

Comment:
The spankey should be stored into span rather than context. Sometimes users may not deliver the correct go context, at that time, the parentContext in `ShouldSuppress` may be `context.Background()`. When the parentContext is `context.Background`, we will use [GLS](https://github.com/alibaba/opentelemetry-go-auto-instrumentation/blob/main/docs/context-propagation.md) to fetch the parent context in this Go Routine, we can fetch the parent span and its attributes in GLS, so you should put the span key into span's attributes rather than the go context.